cmd/ebrelayer/relayer/ethereum.go

```
@@ -127,177 +137,162 @@ func (sub EthereumSub) Start(txFactory tx.Factory, completionEvent *sync.WaitGro
-				if err != nil {
-					sub.SugaredLogger.Errorw("failed to transform from log to event.",
-						errorMessageKey, err.Error())
-					continue
-				}
-				if !isBurnLock {
-					sub.SugaredLogger.Infow("not burn or lock event, continue events.")
-					continue
-				}
-				events = append(events, event)
-			}
-
-			if len(events) > 0 {
-				if err := sub.handleEthereumEvent(txFactory, events); err != nil {
-					sub.SugaredLogger.Errorw("failed to handle ethereum event.",
-						errorMessageKey, err.Error())
-				}
-				time.Sleep(transactionInterval)
-			}
-			// add 1 to the current block so we don't reprocess it
-			endingBlock = endingBlock.Add(endingBlock, big.NewInt(1))
-			// save the current ending block + 1 to the lastprocessed block to ensure we keep reading blocks sequentially and don't repeat blocks
-			err = sub.DB.Put([]byte(ethLevelDBKey), endingBlock.Bytes(), nil)
-			if err != nil {
-				// if you can't write to leveldb, then error out as something is seriously amiss
-				log.Fatalf("Error saving lastProcessedBlock to leveldb: %v", err)
-			}
-			lastProcessedBlock = endingBlock
+		case <-t.C:
+			sub.CheckNonceAndProcess(txFactory, networkID, ethClient, tmClient, bridgeBankAddress, bridgeBankContractABI)
 		}
 	}
 }
 
-func (sub EthereumSub) getAllClaims(fromBlock int64, toBlock int64) []types.EthereumBridgeClaim {
-	log.Printf("Replay get all ethereum bridge claim from block %d to block %d\n", fromBlock, toBlock)
+// CheckNonceAndProcess check the lock burn nonce and process the event
+func (sub EthereumSub) CheckNonceAndProcess(txFactory tx.Factory,
+	networkID *big.Int,
+	ethClient *ethclient.Client,
+	tmClient *tmclient.HTTP,
+	bridgeBankAddress common.Address,
+	bridgeBankContractABI abi.ABI) {
+	// get current block height
+	currentBlock, err := ethClient.HeaderByNumber(context.Background(), nil)
+	if err != nil {
+		return
```

It looks like we're failing silently if these calls fails. Is this the intended behaviour?

---

log added if call failed.